nexus/types/src/external_api/views.rs

```diff
@@ -873,6 +873,32 @@ impl fmt::Display for PhysicalDiskState {
     }
 }
 
+// UPDATES
+
+/// The source of the target release.
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, JsonSchema, Serialize)]
+#[serde(rename_all = "snake_case")]
+pub enum TargetReleaseSource {
+    /// Obtain the release artifact from the `install` dataset.
+    InstallDataset,
+
+    /// Use the specified release of the rack's system software.
+    SystemVersion(SemverVersion),
+}
```

Comment:
Move this into `nexus/types/src/external_api/shared.rs` since you're using it in both views and params. Silly, but that's the scheme.

Reply:
Thank you, moved in `fe23073`.